src/getServerOptions.ts

```diff
@@ -4,21 +4,15 @@ import { JavaConfig } from "./getJavaConfig";
 interface GetServerOptions {
   metalsClasspath: string;
   serverProperties: string[] | undefined;
-  clientName: string;
   javaConfig: JavaConfig;
 }
 
 export function getServerOptions({
   metalsClasspath,
   serverProperties = [],
-  clientName,
```

Comment:
My main question I guess is about removing this, or making it optional. Originally I was thinking of just removing it, but then if we remove this and fully configure with `InitializationOptions`, the next release of the extension won't be configured correctly when using old versions of Metals. Does that matter? Or do we keep this in here as optional and then still pass in the client in `vscode` and `coc-metals` to ensure that it works with old versions of metals, but also include all the `InitializationOptions` to ensure it's correctly configured with the latest. I guess as I'm typing this I'm leaning towards the later.

Reply:
Ok, so I changed this around to just make it optional.

Reply:
Ah sorry, I was from mobile and missed this discussion.
I agree it makes sense to keep it for now, but I would probably remove it at some point.
People can always use and older version of the extension if they really want to.
Or alternatively pass in the client property manually in the options

Reply:
Alright, sounds good. I think I'll keep this for the next release, and then after that remove it. I think this way it will make sure the transition is smooth. Then following that, there should be no reason to be _multiple_ versions behind.